src/AlgebraicGeometry/Schemes/AffineSchemes/Objects/Attributes.jl

```diff
@@ -387,13 +387,33 @@ julia> dim(Y) # one dimension comes from ZZ and two from x1 and x2
 dim(X::AbsSpec)
 
 @attr function dim(X::AbsSpec{<:Ring, <:MPolyQuoLocRing})
-  return dim(saturated_ideal(modulus(OO(X))))
+  error("Not implemented")
+end
```

Comment:
One comment beforehand: Most of this should not happen on the geometric side, but on the algebra side. We need all these fixes on the algebra side as well and should only have them in one spot. The geometric side should directly refer dim (X) to dim(OO(X)) in the local and affine cases.

Reply:
Ok, agreed. It seems that a lot of this is still missing on the algebra side, so this might be a good occasion to add it.

Reply:
How serious are we about doing things on the algebra side? 
If we are serious, then there would be only a single method:
`dim(X::AbsSpec) = dim(OO(X))`
and that is it.  But yea, in the long run this is maybe the best way to go.

Reply:
This is definitely the way to go. Otherwise there will be tons of duplicate code.

(My issue was meant as a marker not to forget this (large and necessary) cleanup. I wrote it on the geometric side because of my own lazyness, as I had stumbled into it on that side. But I always expected the solution to be implemented on the algebraic side and only a simple call on the geometric one.